velero-plugins/pod/restore.go

```diff
@@ -113,9 +113,43 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	destStagePodImage := input.Restore.Annotations[common.StagePodImageAnnotation]
 	if len(pod.Labels[common.IncludedInStageBackupLabel]) > 0 && len(destStagePodImage) > 0 {
 		p.Log.Infof("[pod-restore] swapping stage pod images for pod %s", pod.Name)
+		pvcVolumes := []corev1API.Volume{}
+		exclude := []string{}
+		for _, volume := range pod.Spec.Volumes {
+			if volume.PersistentVolumeClaim == nil {
+				exclude = append(exclude, volume.Name)
+				continue
+			}
+			if len(pod.Labels[common.ExcludePVCPodLabel]) > 0 {
+				if strings.Contains(pod.Labels[common.ExcludePVCPodLabel], volume.Name) {
```

Comment:
We need to split the Exclude label out into a slice and compare slice entries. In the simple case, "foo,bar,baz" does contain "foo" so we'll exclude foo, "foo1,foo2" will contain "foo" so foo would be erroneously skipped.